### ISSUE TITLE
OpenCV Samples compilation with OpenCV binaries does not work

### ISSUE THREAD

**Issue by ilya-lavrenov** [Itseez/opencv#5541]
See https://github.com/Itseez/opencv/blob/master/samples/cpp/CMakeLists.txt#L62

Samples use `OPENCV_LINKER_LIBS` variable to link necessary libraries, but this variable is empty when samples, libs are installed e.g. from deb packages.
So, we have the undefined symbols:
```
Linking CXX executable c-example-bgfg_codebook
Linking CXX executable c-example-blobtrack_sample
Linking CXX executable c-example-adaptiveskindetector
Linking C executable c-example-contours
/usr/bin/ld: CMakeFiles/example_contours.dir/contours.c.o: undefined reference to symbol 'lrint@@GLIBC_2.4'
//lib/arm-linux-gnueabihf/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [c/c-example-contours] Error 1
make[1]: *** [c/CMakeFiles/example_contours.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  1%] [  2%] [  2%] Built target example_bgfg_codebook
Built target example_blobtrack_sample
Built target example_adaptiveskindetector
make: *** [all] Error 2
```

BTW, it's reproducible on U14, but it seems to be working on U12.

**Comment by asmorkalov** [comment]
Non-accurate fix in MR #5544. The best solution is to provide list of dependencies in OpenCV Cmake config.

**Comment by alalek** [comment]
Anyway, it is bad idea to fix samples.

`OPENCV_LINKER_LIBS` may contain:
- system libs: `m dl rt pthread`
- libs via absolute pats: `/usr/lib64/libz.so` or `/lib64/libGLU.so`
- CMake targets, like ippicv / vtk / etc
